packages/@aws-cdk/aws-apigatewayv2/lib/api.ts

```diff
@@ -0,0 +1,186 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { CfnApi, CfnApiProps,  HttpRouteOptions, IRouteBase, LambdaRouteOptions, Route  } from '../lib';
+
+/**
+ * the HTTP API interface
+ */
+export interface IHttpApi extends cdk.IResource {
+  /**
+   * The ID of this API Gateway HTTP Api.
+   * @attribute
+   */
+  readonly httpApiId: string;
+}
+
+abstract class ApiBase extends cdk.Resource implements IHttpApi {
+  /**
+   * API ID
+   */
+  public abstract readonly httpApiId: string;
+  /**
+   * API URL
+   */
+  public abstract readonly url: string;
+}
+
+/**
+ * API properties
+ */
+export interface HttpApiProps extends cdk.StackProps {
+  /**
+   * A name for the HTTP API resoruce
+   *
+   * @default - ID of the HttpApi construct.
+   */
+  readonly apiName?: string;
+  /**
+   * API protocol
+   *
+   * @default HTTP
+   */
+  readonly protocol?: ProtocolType;
+  /**
+   * target lambda function for lambda proxy integration.
+   *
+   * @default - None. Specify either `targetHandler` or `targetUrl`
+   */
+  readonly targetHandler?: lambda.IFunction;
+
+  /**
+   * target URL for HTTP proxy integration.
+   *
+   * @default - None. Specify either `targetHandler` or `targetUrl`
+   */
+  readonly targetUrl?: string;
+}
+
+/**
+ * protocol types for the Amazon API Gateway HTTP API
+ */
+export enum ProtocolType {
+  /**
+   * HTTP protocol
+   */
+  HTTP = 'HTTP',
+  /**
+   * Websocket protocol
+   */
+  WEBSOCKET = 'WEBSOCKET'
+}
+
+/**
+ * HTTPApi Resource Class
+ *
+ * @resource AWS::ApiGatewayV2::Api
+ */
+export class HttpApi extends ApiBase implements IHttpApi {
+  /**
+   * import from ApiId
+   */
+  public static fromApiId(scope: cdk.Construct, id: string, httpApiId: string): IHttpApi {
+    class Import extends cdk.Resource implements IHttpApi {
+      public readonly httpApiId = httpApiId;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * the API identifer
+   */
+  public readonly httpApiId: string;
+  /**
+   * AWS partition either `aws` or `aws-cn`
+   */
+  public readonly partition: string;
+  /**
+   * AWS region of this stack
+   */
+  public readonly region: string;
+  /**
+   * AWS account of this stack
+   */
+  public readonly account: string;
+  /**
+   * AWS domain name either `amazonaws.com` or `amazonaws.com.cn`
+   */
+  public readonly awsdn: string;
```

Comment:
Drop. None of these should be properties of `HttpApi`. They're all properties of the stack.